expression/builtin_string_vec.go

```diff
@@ -2162,11 +2164,67 @@ func (b *builtinBitLengthSig) vecEvalInt(input *chunk.Chunk, result *chunk.Colum
 }
 
 func (b *builtinCharSig) vectorized() bool {
-	return false
+	return true
 }
 
 func (b *builtinCharSig) vecEvalString(input *chunk.Chunk, result *chunk.Column) error {
-	return errors.Errorf("not implemented")
+	n := input.NumRows()
+	l := len(b.args)
+	buf := make([]*chunk.Column, l-1)
+	for i := 0; i < len(b.args)-1; i++ {
+		te, err := b.bufAllocator.get(types.ETInt, n)
+		if err != nil {
+			return err
+		}
+		buf[i] = te
+		defer b.bufAllocator.put(buf[i])
+		if err := b.args[i].VecEvalInt(b.ctx, input, buf[i]); err != nil {
+			return err
+		}
+	}
+	bufstr, err := b.bufAllocator.get(types.ETString, n)
+	defer b.bufAllocator.put(bufstr)
+	if err != nil {
+		return err
+	}
+	if err := b.args[l-1].VecEvalString(b.ctx, input, bufstr); err != nil {
+		return err
+	}
+	result.ReserveString(n)
+
+	for i := 0; i < n; i++ {
+		bigints := make([]int64, 0, l-1)
+		for j := 0; j < l-1; j++ {
+			if buf[j].IsNull(i) {
+				continue
+			}
+			bigints = append(bigints, buf[j].GetInt64(i))
+		}
+		tempres := string(b.convertToBytes(bigints))
+		charsetlabe := strings.ToLower(bufstr.GetString(i))
+		if bufstr.IsNull(i) || charsetlabe == "ascii" || strings.HasPrefix(charsetlabe, "utf8") {
+			result.AppendString(tempres)
+		} else {
+			encoding, charsetName := charset.Lookup(charsetlabe)
+			if encoding == nil {
+				result.AppendNull()
+				continue
```

Comment:
Should we return the error instead of appending null ?

Reply:
Yes.After return err the calculate will be interrupted.I will solve this problem